_job-scheduler-plugin/build-plugin.md

```diff
@@ -0,0 +1,163 @@
+    public Schedule getSchedule() {
+        return this.schedule;
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return this.isEnabled;
+    }
+
+    @Override
+    public Long getLockDurationSeconds() {
+        return this.lockDurationSeconds;
+    }
+
+    @Override public Double getJitter() {
+        return jitter;
+    }
+
+    public String getIndexToWatch() {
+        return this.indexToWatch;
+    }
+
+    public void setJobName(String jobName) {
+        this.jobName = jobName;
+    }
+
+    public void setLastUpdateTime(Instant lastUpdateTime) {
+        this.lastUpdateTime = lastUpdateTime;
+    }
+
+    public void setEnabledTime(Instant enabledTime) {
+        this.enabledTime = enabledTime;
+    }
+
+    public void setEnabled(boolean enabled) {
+        isEnabled = enabled;
+    }
+
+    public void setSchedule(Schedule schedule) {
+        this.schedule = schedule;
+    }
+
+    public void setIndexToWatch(String indexToWatch) {
+        this.indexToWatch = indexToWatch;
+    }
+
+    public void setLockDurationSeconds(Long lockDurationSeconds) {
+        this.lockDurationSeconds = lockDurationSeconds;
+    }
+
+    public void setJitter(Double jitter) {
+        this.jitter = jitter;
+    }
+```
+
+The example code above defines the request parameters that can be set when building your plugin. Reference the following table configured by the example code above.
+
+| Field | Data type | Description |
+:--- | :--- | :---
+| getName | String | Returns the name of the job. |
+| getLastUpdateTime | Time unit | Returns the last time the job ran. |
+| getEnabledTime | Time unit | Returns the time that the job was enabled. |
+| getSchedule | Unix cron | Returns the schedule that the job runs on formatted in Unix cron syntax. |
+| isEnabled | Boolean | Whether or not the job is enabled. |
+| getLockDurationSeconds | Integer | Returns the duration of time that the job is locked. |
+| getJitter | Integer | Returns the defined jitter value. |
+| getIndexToWatch | String | Returns the name of the index that the job monitors. |
+| setJobName | String | Sets the name of the job. |
+| setLastUpdateTime | Time unit | Sets the time that the job was last updated at. |
+| setEnabledTime | Time unit | Sets the time for when the job is enabled. |
+| setEnabled | Boolean | Enables or disables the job. |
+| setSchedule | Unix cron | Sets the job's schedule using Unix cron syntax. |
+| setIndexToWatch | String | Set the index that the job will monitor. |
+| setLockDurationSeconds | Integer | Sets the duration that the job is locked for. |
+| setJitter | Double | Sets the jitter value. |
+
+For further information, reference the Job Scheduler [sample extention](https://github.com/opensearch-project/job-scheduler/blob/main/sample-extension-plugin/src/main/java/org/opensearch/jobscheduler/sampleextension/SampleJobParameter.java) directory in the [Job Scheduler Github repro](https://github.com/opensearch-project/job-scheduler).
```

Comment:
Add a paragraph for JobRunner

The logic run by your job should be defined by a class extending `ScheduledJobRunner` like `SampleJobRunner`
During the job running, there's a locking mechanism you can use to prevent other nodes executes the same job.
acquire: https://github.com/opensearch-project/job-scheduler/blob/main/sample-extension-plugin/src/main/java/org/opensearch/jobscheduler/sampleextension/SampleJobRunner.java#L96
don't forget to release before job finishes: https://github.com/opensearch-project/job-scheduler/blob/main/sample-extension-plugin/src/main/java/org/opensearch/jobscheduler/sampleextension/SampleJobRunner.java#L116